devicetypes/smartthings/arrival-sensor-ha.src/arrival-sensor-ha.groovy

```diff
@@ -58,17 +64,33 @@ metadata {
 }
 
 def updated() {
-    stopTimer()
-    startTimer()
+    log.debug "updated()"
+    if (isVision()) {
+        state.gsensor = 0
+        def thedetectTime = (detectTime ? detectTime as int : 2) * 1
+        def updatecmds = zigbee.writeAttribute(0x0000, 0x0000, 0x20, thedetectTime, [mfgCode: 0x120D])
+        log.debug "Updatecmds:  ${updatecmds}"
+        return response(updatecmds)
+    } else {
+        stopTimer()
+        startTimer()
+    }
 }
 
 def installed() {
+    log.debug "installed()"
     // Arrival sensors only goes OFFLINE when Hub is off
     sendEvent(name: "DeviceWatch-Enroll", value: JsonOutput.toJson([protocol: "zigbee", scheme:"untracked"]), displayed: false)
 }
 
 def configure() {
-    def cmds = zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) + zigbee.batteryConfig(20, 20, 0x01)
+    def cmds = []
+    if (isVision()) {
+        cmds = zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020) + zigbee.batteryConfig(3600, 3600, 0x01) +   //3600 -> 1hour
```

Comment:
since `zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020)` is used in both cases I suggest to move it to `def cmds = zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, 0x0020)`

Reply:
OK! I will modify the code according to suggestion.